File: models/Post.kt
package com.whatalab.vipspad.models

import java.util.*

data class Post(
    val body: String,
    val category: Category,
    val category_id: Int,
    val created_at: Date,
    val creator: Creator,
    val creator_id: Int,
    val discord_url: String,
    val id: Int,
    val image2_url: String,
    val image_url: String,
    val market_url: String,
    val network: Network,
    val network_id: Int,
    val risk_id: Int,
    val risk_index: RiskIndex,
    val title: String
)

File: Vipspad/Vipspad/Classes/Common/Entities/External/Post.swift
//
//  Post.swift
//  Vipspad
//
//  Created by Azure on 8/1/23.
//

import Foundation

struct Post: Codable {
    let id: Int
    let title: String
    let body: String
    let discordURL: String
    let imageURL: String
    let image2URL: String
    let marketURL: String
    let creatorID: Int
    let categoryID: Int
    let networkID: Int
    let riskID: Int
    let createdAt: String
    let creator: BaseProp
    let network: BaseProp
    let riskIndex: BaseProp
    let category: BaseProp

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case body
        case discordURL = "discord_url"
        case imageURL = "image_url"
        case image2URL = "image2_url"
        case marketURL = "market_url"
        case creatorID = "creator_id"
        case categoryID = "category_id"
        case networkID = "network_id"
        case riskID = "risk_id"
        case createdAt = "created_at"
        case creator
        case network
        case riskIndex = "risk_index"
        case category
    }
}

typealias Posts = [Post]
